Add pagination by cursor (#168)
Scalingo/go-utils: mongo/pagination/README.md

The returned meta object contains pagination metadata that could be used in the
request answer.

## Pagination by Cursor

This package also provide a pagination by cursor, could be useful in the case of
frequently updated data.

> It follows the mongoDB pagination using range queries, explained on

Reviewer: ```suggestion
It follows the MongoDB pagination using range queries, explained on
```

[their documentation](https://docs.mongodb.com/manual/reference/method/cursor.skip/#using-range-queries).

This pagination takes a bson object as cursor. This bson determines the field

> and the comparaison of mongo document in accordance with the order.

Reviewer: ```suggestion
and the comparison of MongoDB document in accordance with the order.
```

For example, in case of ordered query: to get the next page we must give a
comparison greater than the last returned value. So the bson should look like this:
```Go
bson.M{"app_id": bson.M{"$gt": 10}}
```
Where `10` is the last `app_id` found.

### How to use it

First, set the default value of number of items per page and the maximum of item

> by page as the following snippet:

Reviewer: ```suggestion
per page as the following snippet:
```

```Go
pageService := pagination.NewPaginationByCursorService(pagination.ServiceOpts{
    PerPageDefault: 5,
    MaxPerPage:     15,
})
```

Then call the `PaginateByCursor` described bellow:

```Go
// resultObject will be filled with the query result.
resultObject := []*ResultObject{}
dbQuery := bson.M{"searched_field": "field_1"}

paginateOpts := PaginateByCursorOpts{

>     AmountItems 5       // Amount of items by page (can be empty)
>     Query       dbQuery // Query which will be executed on the database (can be nil)
>     SortOrder   "-_id"  // The field for the sort order (by default "-_id")

Reviewer: ```suggestion
    AmountItems: 5       // Amount of items by page (can be empty)
    Query:       dbQuery // Query which will be executed on the database (can be nil)
    SortOrder:   "-_id"  // The field for the sort order (by default "-_id")
```

}

err := pageService.PaginateByCursor(
    ctx,                    // A context (required)
    "RequestedCollection",  // Name of the collection (required)
    &resultObject,          // The object that will contain the data (must be an array)
    paginateOpts)
// Here `resultObject` contains the first page

// To get the next page, you must provide the cursor in the `PaginateByCursorOpts`
// object, in accordance with the order of the data.
// The following is an example to retrieve the second page in a case of reverse ordered data.
resultObjectLen := len(resultObjectLen)
cursor := bson.M{"_id": bson.M{"$lt": resultObject[resultObjectLen-1].ID}}

paginateOpts := PaginateByCursorOpts{

>     Cursor      cursor  // The bson cursor, it must provide the comparison in accordance with the order (can be nil)
>     AmountItems 5
>     Query       dbQuery
>     SortOrder   "-_id"

Reviewer: ```suggestion
    Cursor:      cursor  // The bson cursor, it must provide the comparison in accordance with the order (can be nil)
    AmountItems: 5
    Query:       dbQuery
    SortOrder:   "-_id"
```
